src/audio/pipeline.c

```diff
@@ -364,11 +368,61 @@ static int pipeline_comp_hw_params(struct comp_dev *current,
 	return 0;
 }
 
+static int pipeline_comp_params_neg(struct comp_dev *current,
+				    struct comp_buffer *calling_buf,
+				    struct pipeline_walk_context *ctx,
+				    int dir)
+{
+	struct pipeline_data *ppl_data = ctx->comp_data;
+	uint32_t flags = 0;
+	int err = 0;
+
+	pipe_cl_dbg("pipeline_comp_params_neg(), current->comp.id = %u, dir = %u",
```

Comment:
It runs in the `ppl_data->p` context, so use `pipe_dbg(ppl_data->p, "....", ...)` to include the pipeline id in the trace messages.

Reply:
ok, I just followed the existed ones in pipeline_comp_params() and pipeline_comp_hw_params(), fwiw, let me change it to pipe_dbg.